### PR TITLE
CI + Performance fixes

### PR DESCRIPTION
**Pull request by Nasfame**

### CODE REVIEW

**Comment by Nasfame** [comment]
@bxcodec i think we can status checks run without the review !!!

File: CODEOWNERS [hunk 1]
```diff
@@ -1 +1 @@
-* @bxcodec @goferHiro
+* @bxcodec @goferHiro @Nasfame
```
Comment (bxcodec) [inline CODEOWNERS:1]: Can you remove the old username, please?

**Comment by bxcodec** [comment]
There's a failed test

**Comment by Nasfame** [comment]
> There's a failed test

i will look into it

**Comment by Nasfame** [comment]
> > There's a failed test
> 
> i will look into it

Fuzzing failed to due cpu overhead

**Comment by Nasfame** [comment]
@bxcodec the tests are fixed

**Comment by Nasfame** [comment]
@bxcodec  can u approve the workflows

**Comment by Nasfame** [comment]
@bxcodec ..?

**Comment by Nasfame** [comment]
Adding proposed solution in #20 for global context to avoid unnecessary function calls. 

Since context.Background is returning a global variable its fine to use backgroundCtx

**Pull request closed without merging**